app/src/main/java/cloudcity/util/CellUtil.java

```diff
@@ -0,0 +1,124 @@
+package cloudcity.util;
+
+import androidx.annotation.NonNull;
+
+import java.util.List;
+import java.util.Objects;
+
+import cloudcity.networking.models.MeasurementsModel;
+import de.fraunhofer.fokus.OpenMobileNetworkToolkit.DataProvider.CellInformations.CellInformation;
+import de.fraunhofer.fokus.OpenMobileNetworkToolkit.DataProvider.CellInformations.LTEInformation;
+import de.fraunhofer.fokus.OpenMobileNetworkToolkit.DataProvider.CellInformations.NRInformation;
+import de.fraunhofer.fokus.OpenMobileNetworkToolkit.DataProvider.DataProvider;
+
+public class CellUtil {
+
+    /**
+     * Finds a registered cell information in a list of cell informations
+     * @param cellList
+     * @return the first registered cell in the list, or null if no registered cells were found
+     */
+    public static CellInformation findRegisteredCell(@NonNull List<CellInformation> cellList) {
+        CellInformation retVal = null;
+
+        for (CellInformation ci: cellList) {
+            if (!ci.isRegistered()) {
+                continue;
+            }
+
+            retVal = ci;
+        }
+
+        return retVal;
+    }
```

Comment:
we don't care if it's the first or the last, we just care that it's registered. And since there's a strong possibility that only one of them can be registered at a time...

Reply:
will not change, resolving.